lib/events/athena/publisher.go

```diff
@@ -94,11 +158,28 @@ func newPublisherFromAthenaConfig(cfg Config) *publisher {
 		so.MaxAttempts = 20
 		so.MaxBackoff = 1 * time.Minute
 	})
-	return NewPublisher(PublisherConfig{
-		TopicARN: cfg.TopicARN,
-		SNSPublisher: sns.NewFromConfig(*cfg.PublisherConsumerAWSConfig, func(o *sns.Options) {
+	hc := awshttp.NewBuildableClient().WithTransportOptions(func(t *http.Transport) {
+		// attempt to aggressively reuse connections for the sake of
+		// avoiding TLS handshakes (the default MaxIdleConnsPerHost is a
+		// pitiful 2)
+		t.MaxIdleConns = 10000
+		t.MaxIdleConnsPerHost = 10000
+	})
+	var messagePublisher messagePublisherFunc
+	if cfg.TopicARN == "" {
+		messagePublisher = SQSPublisherFunc(cfg.QueueURL, sqs.NewFromConfig(*cfg.PublisherConsumerAWSConfig, func(o *sqs.Options) {
+			o.Retryer = r
+			o.HTTPClient = hc
+		}))
+	} else {
+		messagePublisher = SNSPublisherFunc(cfg.TopicARN, sns.NewFromConfig(*cfg.PublisherConsumerAWSConfig, func(o *sns.Options) {
 			o.Retryer = r
-		}),
+			o.HTTPClient = hc
+		}))
+	}
```

Comment:
nit: 

Can we move creation of `messagePublisher` to `NewPublisher` ?

Reply:
Hm, how would you do that? Pass both `TopicARN` and `QueueURL` in `PublisherConfig`? That's not going to be fun for tests :\

Reply:
`Yes, Ideally the NewPublisher should take care of of creation of `messagePublisher `based on appropriate config but if this change will require a lot of test alignment I will not persist.